src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/Internal/Win32.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Hosting.WindowsServices.Internal
+{
+    internal static class Win32
```

Comment:
For this PR it is probably best to bring it over as-is. But we should follow up with moving these into the typical "Interop" pattern we have established in dotnet/runtime/src/libraries.

Reply:
> For this PR it is probably best to bring it over as-is. But we should follow up with moving these into the typical "Interop" pattern we have established in dotnet/runtime/src/libraries

yes thats the plan